### PR TITLE
Tms pipeline fb90016 fb90046 fb89749 fb90015 fb90046

### PR DESCRIPTION
**Pull request by TimSVector**

### CODE REVIEW

File: src/main/java/com/vectorcast/plugins/vectorcastexecution/job/NewPipelineJob.java [hunk 1]
```diff
@@ -329,6 +334,38 @@ private String getBaseJenkinsfile() throws IOException {
 	 * @return script portion of pipeline job.
 	 * @throws IOException
 	 */
+    public String getVersion() {
+        
+        String path = null;
```
Comment (nbelliot) [inline src/main/java/com/vectorcast/plugins/vectorcastexecution/job/NewPipelineJob.java:339]: Can we initialize this variable inside the `try` block, like this:
`String path = getClass().getProtectionDomain().getCodeSource().getLocation().getPath();`
That way we reduce the variable's scope and can avoid setting it to null.

File: src/main/java/com/vectorcast/plugins/vectorcastexecution/VectorCASTSetup.java [hunk 2]
```diff
@@ -429,6 +446,33 @@ private void processDir(File dir, String base, FilePath destDir, Boolean directD
             }
         }
     }
+    
+    public void printVersion(TaskListener listener, JarFile jFile) {
```
Comment (nbelliot) [inline src/main/java/com/vectorcast/plugins/vectorcastexecution/VectorCASTSetup.java:450]: Looks like this contains duplicate code as `NewPipelineJob::getVersion`.
Can we move `NewPipelineJob::getVersion` outside of `NewPipelineJob` and have `NewPipelineJob` and this function call it?

File: src/main/resources/scripts/generate-results.py [hunk 3]
```diff
@@ -257,7 +257,8 @@ def genDataApiReports(entry, jUnit, cbtDict):
             
     except Exception as e:
         print("ERROR: failed to generate XML reports using vpython and the Data API for ", entry["compiler"] + "_" + entry["testsuite"] + "_" + entry["env"], "in directory", entry["build_dir"])
-        traceback.print_exc()
+        if verbose:
```
[the remark this answers is not in the document]
Comment (TimSVector) [reply src/main/resources/scripts/generate-results.py:260]: In the case of a disabled Compiler/TestSuite/Environment in a manage project, the report cannot be generated and there's no way to indicate from a manage -build-directory-data command as to which environments are disabled.  So you end up getting a false failing when really its only disabled.  The println will still happen, I just don't want to step/stage to show up as failed.

File: src/main/java/com/vectorcast/plugins/vectorcastexecution/job/NewPipelineJob.java (hunk 1 above)
Comment (TimSVector) [reply src/main/java/com/vectorcast/plugins/vectorcastexecution/job/NewPipelineJob.java:339]: If you can find a better/safer way to implement, that's awesome.  Please note that there is a similar block inside VectorCASTSetup.java.

File: src/main/java/com/vectorcast/plugins/vectorcastexecution/VectorCASTSetup.java (hunk 2 above)
Comment (TimSVector) [reply src/main/java/com/vectorcast/plugins/vectorcastexecution/VectorCASTSetup.java:450]: If you can find a better/safer way to implement, that's awesome. Please note that there is a similar block inside NewPipelineJob.java.

File: src/main/java/com/vectorcast/plugins/vectorcastexecution/job/NewPipelineJob.java (hunk 1 above)
Comment (TimSVector) [reply src/main/java/com/vectorcast/plugins/vectorcastexecution/job/NewPipelineJob.java:339]: I was trying keep everything as localized as possible as I don't know what side effects there would be in trying to have VectorCASTSetup.java access something in NewPipelineJob.java and vice versa.  I also wasn't sure if there was a place both could access (BaseJob.java?) that would have access to the right data at the right time.

**Pull request merged**